Source/Microsoft.Teams.Apps.CompanyCommunicator.Data.Func/CompanyCommunicatorDataFunction.cs

```diff
@@ -0,0 +1,86 @@
+    using Microsoft.Teams.Apps.CompanyCommunicator.Data.Func.Services.NotificationDataServices;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Azure Function App triggered by messages from a Service Bus queue
+    /// Used for incrementing results for a sent notification.
+    /// </summary>
+    public class CompanyCommunicatorDataFunction
+    {
+        private readonly NotificationDataRepository notificationDataRepository;
+        private readonly ForceCompleteNotificationDataService forceCompleteNotificationDataService;
+        private readonly UpdateCountsInNotificationDataService updateCountsInNotificationDataService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompanyCommunicatorDataFunction"/> class.
+        /// </summary>
+        /// <param name="notificationDataRepository">The notification data repository.</param>
+        /// <param name="forceCompleteNotificationDataService">The force complete notification data service.</param>
+        /// <param name="updateCountsInNotificationDataService">The update counts in notification data service.</param>
+        public CompanyCommunicatorDataFunction(
+            NotificationDataRepository notificationDataRepository,
+            ForceCompleteNotificationDataService forceCompleteNotificationDataService,
+            UpdateCountsInNotificationDataService updateCountsInNotificationDataService)
+        {
+            this.notificationDataRepository = notificationDataRepository;
+            this.forceCompleteNotificationDataService = forceCompleteNotificationDataService;
+            this.updateCountsInNotificationDataService = updateCountsInNotificationDataService;
+        }
+
+        /// <summary>
+        /// Azure Function App triggered by messages from a Service Bus queue
+        /// Used for aggregating results for a sent notification.
+        /// </summary>
+        /// <param name="myQueueItem">The Service Bus queue item.</param>
+        /// <param name="deliveryCount">The deliver count.</param>
+        /// <param name="enqueuedTimeUtc">The enqueued time.</param>
+        /// <param name="messageId">The message ID.</param>
+        /// <param name="log">The logger.</param>
+        /// <param name="context">The execution context.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        [FunctionName("CompanyCommunicatorDataFunction")]
+        public async Task Run(
+            [ServiceBusTrigger(
+                DataQueue.QueueName,
+                Connection = DataQueue.ServiceBusConnectionConfigurationKey)]
+            string myQueueItem,
+            int deliveryCount,
+            DateTime enqueuedTimeUtc,
+            string messageId,
+            ILogger log,
+            ExecutionContext context)
+        {
+            var messageContent = JsonConvert.DeserializeObject<DataQueueMessageContent>(myQueueItem);
```

Comment:
Please add logging to this Azure function.
e.g., when we force the completion, and when we update the counts